library/x509_crt.c

```diff
@@ -855,6 +882,12 @@ static int x509_get_basic_constraints( unsigned char **p,
     if( *max_pathlen == INT_MAX )
         return( MBEDTLS_ERR_ASN1_INVALID_LENGTH );
 
+    /* Do not accept max_pathlen equal to INT_MAX to avoid a signed integer
+     * overflow, which is an undefined behavior. */
+    if( *max_pathlen == INT_MAX )
```

Comment:
Why do we need both checks line 882 and line 887 ?

Reply:
Good catch. This fix was merged both into baremetal and the mainline mbedtls in a different form, and somehow git didn't pick this up as a conflict - just merged both of them. The old version should stay here, new one is dead code. I'll fix that!